pkg/injector/iptables.go

```diff
@@ -74,6 +74,11 @@ func generateIptablesCommands(outboundIPRangeExclusionList []string, outboundIPR
 	// 1. Create inbound rules
 	cmds = append(cmds, iptablesInboundStaticRules...)
 
+	// Ignore inbound traffic on specified interfaces
+	for _, iface := range networkInterfaceExclusionList {
+		cmds = append(cmds, fmt.Sprintf("-I OSM_PROXY_INBOUND -i %s -j RETURN", iface))
```

Comment:
What's the reason the inbound rule uses -I (insert before) and outbound rule uses -A (append)?

If ordering is important, kindly document this in the code.

Reply:
These inbound rules only work when they're added before the static rules, either with `-I` or by adding them to the slice before with `-A`. I would need help figuring out an exact reason why that's the case.

Reply:
Iptable rules are executed in order. So it makes sense to use `-I` here so that the exclusions get applied first before other rules that can match the traffic. Could you add a comment here and also for `-I OSM_PROXY_INBOUND -p tcp --match multiport --dports %s -j RETURN` which is missing a comment.

I am looking into the code to see if the ordering of outbound rules is correct.

Reply:
I narrowed it down to needing these commands at least before the last static rule: 
```go
	// Redirect remaining inbound traffic to Envoy
	"-A OSM_PROXY_INBOUND -p tcp -j OSM_PROXY_IN_REDIRECT",
```

Reply:
So this is how the rules should be ordered.

INBOUND:
exclusions
inclusions
static rules that don't redirect to proxy
redirect to proxy

OUTBOUND
exclusions
inclusions
status rules that don't redirect to proxy
redirect to proxy

The reason `-I` is used with inbound is because the `redirect to proxy rule: ...-A OSM_PROXY_INBOUND -p tcp -j OSM_PROXY_IN_REDIRECT` is a part of the static rule set. While for outbound, `redirect to proxy rule: ... -A OSM_PROXY_OUTBOUND -j OSM_PROXY_OUT_REDIRECT` is added only after the exclusions/inclusions rules because the redirection rule OSM_PROXY_OUTBOUND -> OSM_PROXY_OUT_REDIRECT is different when inclusions are applied.

Your change looks good, could you add the following comment next to the inbound rule added:
```
// *Note: it is important to use the insert option '-I' instead of the append option '-A' to ensure the 
// exclusion of traffic to the network interface happens before the rule that redirects traffic to the proxy 
```